integration-tests/metrics-init.patch

```diff
@@ -0,0 +1,134 @@
+--- a/metrics-caffeine3/pom.xml
++++ b/metrics-caffeine3/pom.xml
+@@ -35,6 +35,13 @@
+                 <groupId>com.github.ben-manes.caffeine</groupId>
+                 <artifactId>caffeine</artifactId>
+                 <version>3.1.8</version>
++                <!-- To resolve version conflict flagged by the `maven-enforcer-plugin`. -->
++                <exclusions>
++                    <exclusion>
++                        <groupId>com.google.errorprone</groupId>
++                        <artifactId>error_prone_annotations</artifactId>
++                    </exclusion>
++                </exclusions>
+             </dependency>
+             <dependency>
+                 <groupId>org.checkerframework</groupId>
+--- a/metrics-core/src/test/java/io/dropwizard/metrics5/CsvReporterTest.java
++++ b/metrics-core/src/test/java/io/dropwizard/metrics5/CsvReporterTest.java
+@@ -231,6 +231,7 @@ public class CsvReporterTest {
+         return builder.toString();
+     }
+ 
++    @SuppressWarnings("FilesReadStringWithCharset")
+     private String fileContents(String filename) throws IOException {
+         return new String(Files.readAllBytes(new File(dataDirectory, filename).toPath()), StandardCharsets.UTF_8);
+     }
```

Comment:
If I run it without this, it stops the build over an error; saying it can't find this symbol. If I open it in IntelliJ it complains about the language level of the project 🤔:

```
Usage of API documented as @since 11+ 
```

Which is weird as we simply run it with 11 and 17 without problems. Maybe if we simply override the version in the correct places in the pom it will run fine. 

Ahh yeah that will fix it:
```
[INFO] ----------------< io.dropwizard.metrics5:metrics-core >-----------------
[INFO] Building Metrics5 Core 5.0.0-rc22                                 [5/39]
[INFO]   from metrics-core/pom.xml
[INFO] -------------------------------[ bundle ]-------------------------------
[INFO] 
[INFO] --- enforcer:3.5.0:enforce (enforce) @ metrics-core ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.dependency.DependencyConvergence passed
[INFO] 
[INFO] --- jacoco:0.8.12:prepare-agent (prepare-agent) @ metrics-core ---
[INFO] argLine set to -javaagent:/Users/rick/.m2/repository/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/target/jacoco.exec
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ metrics-core ---
[INFO] skip non existing resourceDirectory /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ metrics-core ---
[INFO] Compiling 11 source files with javac [forked debug parameters release 8] to target/classes
[WARNING] [options] source value 8 is obsolete and will be removed in a future release
[WARNING] [options] target value 8 is obsolete and will be removed in a future release
[WARNING] [options] To suppress warnings about obsolete options, use -Xlint:-options.
[WARNING] /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/main/java/io/dropwizard/metrics5/MetricRegistry.java:[56,27] [this-escape] possible 'this' escape before subclass is fully initialized
[WARNING] /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/main/java/io/dropwizard/metrics5/WeightedSnapshot.java:[68,17] [FormatStringConcatenation] Defer string concatenation to the invoked method
    (see https://error-prone.picnic.tech/bugpatterns/FormatStringConcatenation)
  Did you mean '"%s is not in [0..1]", quantile);'?
[WARNING] /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/main/java/io/dropwizard/metrics5/LockFreeExponentiallyDecayingReservoir.java:[240,19] [FormatStringConcatenation] Defer string concatenation to the invoked method
    (see https://error-prone.picnic.tech/bugpatterns/FormatStringConcatenation)
  Did you mean 'value > 0, "LockFreeExponentiallyDecayingReservoir size must be positive: %s", value);'?
[WARNING] /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/main/java/io/dropwizard/metrics5/UniformSnapshot.java:[51,17] [FormatStringConcatenation] Defer string concatenation to the invoked method
    (see https://error-prone.picnic.tech/bugpatterns/FormatStringConcatenation)
  Did you mean '"%s is not in [0..1]", quantile);'?
[INFO] 
[INFO] --- resources:3.3.1:testResources (default-testResources) @ metrics-core ---
[INFO] skip non existing resourceDirectory /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/test/resources
[INFO] 
[INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ metrics-core ---
[INFO] Compiling 18 source files with javac [forked debug parameters release 8] to target/test-classes
[INFO] -------------------------------------------------------------
[WARNING] COMPILATION WARNING : 
[INFO] -------------------------------------------------------------
[WARNING] [options] source value 8 is obsolete and will be removed in a future release
[WARNING] [options] target value 8 is obsolete and will be removed in a future release
[WARNING] [options] To suppress warnings about obsolete options, use -Xlint:-options.
[INFO] 3 warnings 
[INFO] -------------------------------------------------------------
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/rick/repos/error-prone-support/integration-tests/.repos/metrics/metrics-core/src/test/java/io/dropwizard/metrics5/CsvReporterTest.java:[200,16] error: cannot find symbol
  symbol:   method readString(Path,Charset)
  location: class Files
```

Reply:
Will push a fix + sync in a bit :)